.github/workflows/actions/get-core-dependencies/action.yml

```diff
@@ -0,0 +1,24 @@
+name: 'Get Core Dependencies'
+description: 'initializes core dependencies after repo has been pulled'
+runs:
+  using: composite
+  steps:
+    - name: Use Node 16
+      uses: actions/setup-node@56337c425554a6be30cdef71bf441f15be286854 # v3.1.1
+      with:
+        node-version: 16
+        cache: 'npm'
+
+    - name: Cache Root Node Modules
+      id: get-stencil-root-node-modules-cache
+      uses: actions/cache@48af2dc4a9e8278b89d7fa154b955c30c6aaab09 # v3.0.2
+      env:
+        cache-name: stencil-root-node-modules
+      with:
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('./package-lock.json') }}
+        path: ./node_modules
```

Comment:
Note that `actions/cache` [recommends](https://github.com/actions/cache/blob/main/examples.md#node---npm) not caching `node_modules`.

Reply:
based on https://github.com/actions/setup-node#caching-packages-dependencies it looks like we should already be getting some dependency caching in our `actions/setup-node` use above (b/c `cache: 'npm'`) - so we might be able to just inline that in each of the testing jobs

Reply:
I'll play around with this a bit in a "throwaway PR" (as to not spam y'all) and report back. I agree we probably shouldn't cache `node_modules/` - I think where I want this to move is that if we're running on Linux/Mac we cache `~/.npm` and if we're on windows we cache the equivalent dir.  I'll play with it

Reply:
Fixed in https://github.com/ionic-team/stencil/pull/3332/commits/3bb286922bc1131e480ca37fb053d38365689b3c - I had to add an extra step for windows to ensure that we got the correct `APP_DATA` dir

Reply:
I moved the impl to reusable workflows in https://github.com/ionic-team/stencil/pull/3332/commits/d7a0795dae7bbda3211e7061847e947914445158. I'll write something up about the trade-offs in the morning